docusaurus/docs/iOS/customization/components.md

```diff
@@ -46,53 +46,62 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
-/// if you provide a complete layout for all subviews.
 func setUpLayout()
 
 /// Main point of customizing the way the view updates its content.
-///
-/// **It's called every time view's content changes.** Calling super is recommended but not required if you update
-/// the content of all subviews of the view.
 func updateContent()
 ```
 
-In addition to this, view components follow a clear naming convention and they all have the content being rendered stored as a property called `content`.
+### `setUp()`
+You can see this lifecycle method as a custom constructor of the view since it is only called once in the lifecycle of the component. Here it is a good place for setting delegates, adding gesture recognizers or adding any kind of target action. Usually you want to call `super.setUp()` when overriding this lifecycle method, but you can chose not to if you want to configure all the delegates and actions from scratch.
 
-For instance the `MessageContent` component `content`'s property holds the `ChatMessage` object.
+### `setUpAppearance()`
+This lifecycle is where you can customize the appearance of the component, like changing colors, corner radius, everything that changes the style of the UI but not the layout. You should call `super.setUpAppearance()` if you only want to override some of the view's appearance and not everything.
 
+### `setUpLayout()`
+Here is where you should do customize the layout of the component, for example, changing the position of the views, padding, margins or even remove some child views. All the UI Components of the SDK use **AutoLayout** to layout the views, but our SDK provides a `ContainerStackView` component to make the customization easier. The `ContainerStackView` works very much like a `UIStackView`, in fact, it has almost the same API, but it is better suitable for our needs in terms of customizability. Just like the other lifecycle methods, you can call `super.setUpLayout()` depending on if you want to make the layout of the component from scratch or just want to change some parts of the component.
 
-### Example: custom avatar
+### `updateContent()`
+Finally, this last lifecycle is called whenever the data of the component changes. Here is where you can change the logic of the component, change how the data is displayed or formatted. In the Stream SDK all of the components have a `content` propery that represents the data of the component. The rule of thumb to use this lifecycle is that if the change you want to do depends on the data of the component, then you should use this lifecycle method, even, for example, to do layout changes that are impacted by the content.
 
-Let's say, we want to change the appearance of avatars from the rounded ones to the rectangular ones:
+In addition to this, view components expose their content with the `content` property. For instance the `MessageContent` component `content`'s property holds the `ChatMessage` object.
```

Comment:
I would add some note that if you need to force update content to reflect some changes, you need to call `updateContentIfNeeded()` instead od calling `updateContent()` directly, that `updateContentIfNeeded` ensures that the view is added to superView or viewController is already loaded.